cpp/include/cudf/aggregation.hpp

```diff
@@ -259,16 +261,64 @@ std::unique_ptr<Base> make_collect_list_aggregation(
  * of the list rows.
  *
  * @param null_handling Indicates whether to include/exclude nulls during collection
- * @param nulls_equal   Flag to specify whether null entries within each list should be considered
- * equal
- * @param nans_equal    Flag to specify whether NaN values in floating point column should be
- * considered equal
+ * @param nulls_equal Flag to specify whether null entries within each list should be considered
+ *        equal.
+ * @param nans_equal Flag to specify whether NaN values in floating point column should be
+ *        considered equal.
  */
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_collect_set_aggregation(null_policy null_handling = null_policy::INCLUDE,
                                                    null_equality nulls_equal = null_equality::EQUAL,
                                                    nan_equality nans_equal = nan_equality::UNEQUAL);
 
+/**
+ * @brief Factory to create a MERGE_LISTS aggregation.
+ *
+ * This aggregation is similar to `COLLECT_LIST` with the following differences:
+ *  - It requires the input values to be a lists column, and
+ *  - The value (lists) corresponding to the same key will not result in a list of lists as output
+ *    from `COLLECT_LIST`. Instead, those lists will result in a list generated by merging them
+ *    together.
+ *
+ * In practice, this aggregation is used to merge the partial results of multiple (distributed)
+ * groupby `COLLECT_LIST` aggregations into a final `COLLECT_LIST` result. Those distributed
+ * aggregations were executed on different values columns partitioned from the original values
+ * column, then their results were (vertically) concatenated before given as the values column for
+ * this aggregation.
+ *
+ * Note that this aggregation does not accept any null handling parameter as it is designed for
+ * simply merging lists of the input lists column. As such, the input lists column is required to
+ * NOT be nullable.
```

Comment:
I don't understand why this can't handle nullable input list columns.

Reply:
Technically it can. But it is designed only to merge the results of collect_list/collect_set, which are not nullable. Thus, if its input is nullable then obviously it is operating on a wrong input data.

Reply:
I'll add more comments to the doxygen to clarify this.

Reply:
Why can't `collect_list` be nullable? If a group has a nullable element in it, won't the resulting list have a null element?

Reply:
No, the output will be lists collecting the elements. If we have nulls and:
* If the null handling parameter is `INCLUDED`: the output will be lists like `[null, ....null, ....]`
* If the null handling parameter is `EXCLUDED`: the output will be empty lists.